spec/reek/configuration/app_configuration_spec.rb

```diff
@@ -13,72 +14,61 @@
+      { Reek::Smells::IrresponsibleModule => { 'enabled' => false } }
+    end
+    let(:expected_directory_directives) do
+      { Pathname.new('spec/samples/three_clean_files') =>
+        { Reek::Smells::UtilityFunction => { 'enabled' => false } } }
     end
-  end
 
-  describe '.configure_smell_repository' do
-    it 'should configure a given smell_repository' do
-      described_class.load_from_file(sample_configuration_path)
-      smell_repository = Reek::Smells::SmellRepository.new('def m; end')
-      described_class.configure_smell_repository smell_repository
+    it 'properly loads configuration and processes it' do
+      finder = Reek::Configuration::ConfigurationFileFinder
+      allow(finder).to receive(:find_by_cli).and_return full_configuration_path
 
-      expect(smell_repository.detectors[Reek::Smells::DataClump]).to be_enabled
-      expect(smell_repository.detectors[Reek::Smells::UncommunicativeVariableName]).
-        not_to be_enabled
-      expect(smell_repository.detectors[Reek::Smells::UncommunicativeMethodName]).not_to be_enabled
+      expect(subject.exclude_paths).to eq(expected_exclude_paths)
+      expect(subject.default_directive).to eq(expected_default_directive)
+      expect(subject.directory_directives).to eq(expected_directory_directives)
     end
   end
 
-  describe '.exclude_paths' do
-    let(:config_path) do
-      SAMPLES_PATH.join('configuration/with_excluded_paths.reek')
-    end
+  describe '#directory_directive_for_source' do
+    context 'returns something falsey' do
+      it 'for sources from IO' do
+        expect(subject.directory_directive_for_source(Reek::Source::SourceCode::IO_IDENTIFIER)).
+          to be_falsey
+      end
 
-    it 'should return all paths to exclude' do
-      with_test_config(config_path) do
-        expect(described_class.exclude_paths).to eq [
-          SAMPLES_PATH.join('source_with_exclude_paths/ignore_me'),
-          SAMPLES_PATH.join('source_with_exclude_paths/nested/ignore_me_as_well')
-        ]
+      it 'for sources from strings' do
+        expect(subject.directory_directive_for_source(Reek::Source::SourceCode::STRING_IDENTIFIER)).
+          to be_falsey
       end
     end
-  end
 
-  describe '.load_from_file' do
-    it 'loads the configuration from given file' do
-      described_class.load_from_file(sample_configuration_path)
-      expect(described_class.configuration).to eq(sample_configuration_loaded)
-    end
-  end
+    context 'our source is in a directory for which we have a directive' do
+      let(:baz_config)  { { Reek::Smells::IrresponsibleModule => { enabled: false } } }
+      let(:bang_config) { { Reek::Smells::Attribute => { enabled: true } } }
 
-  describe '.reset' do
-    it 'resets the configuration' do
-      described_class.load_from_file(sample_configuration_path)
+      let(:directory_directives) do
+        {
+          Pathname.new('foo/bar/baz')  => baz_config,
+          Pathname.new('foo/bar/bang') => bang_config
```

Comment:
Whatever comes first for "detect" here: https://github.com/troessner/reek/pull/593/files#diff-5a8a1c2b1f6f3e41dfe85414f9c43266R69
Excellent question. Wondering how we could solve this.
How about adapting the linked code to not just use plain pattern matching but to look for the closest pattern?